ipa-core/src/net/client/mod.rs

```diff
@@ -384,6 +384,31 @@ impl<F: ConnectionFlavor> IpaHttpClient<F> {
         let resp = self.request(req).await?;
         resp_ok(resp).await
     }
+
+    /// This API is used by leader shards in MPC to request query status information on peers.
+    /// If a given peer has status that doesn't match the one provided by the leader, it responds
+    /// with 412 error and encodes its status inside the response body. Otherwise, 200 is returned.
+    ///
+    /// # Errors
+    /// If the request has illegal arguments, or fails to be delivered
+    pub async fn status_match(&self, data: CompareStatusRequest) -> Result<(), Error> {
```

Comment:
1) Can you please add tests to this code? I would be good to have some checks around the parsing of the error and the status.

2) Does this belong in `IpaHttpClient<Shard>` instead of the generic `IpaHttpClient<F>` ?